src/pages/ExamsPage/components/AllowanceListActions.jsx

```diff
@@ -1,17 +1,149 @@
 import { PropTypes } from 'prop-types';
 import {
-  ActionRow,
+  ActionRow, Alert,
   Button,
+  Form,
   Icon,
   IconButtonWithTooltip,
   ModalDialog,
   useToggle,
 } from '@openedx/paragon';
-import { DeleteOutline, EditOutline } from '@openedx/paragon/icons';
+import { DeleteOutline, EditOutline, Info } from '@openedx/paragon/icons';
 import { useIntl } from '@edx/frontend-platform/i18n';
 
+import React, { useState } from 'react';
 import messages from '../messages';
-import { useDeleteAllowance } from '../hooks';
+import { useDeleteAllowance, useEditAllowance, useAllowancesData } from '../hooks';
+import { useClearRequest, useRequestError } from '../../../data/redux/hooks';
+import * as constants from '../../../data/constants';
+
+const EditModal = (isOpen, close, allowance, formatMessage) => {
+  const editAllowance = useEditAllowance();
+  const {
+    allowancesList,
+  } = useAllowancesData();
+
+  console.log('**** allowance list ***** ', allowancesList);
+
+  const allowances = allowancesList.filter(a => a.id === allowance.id);
```

Comment:
can you clarify why this is needed? It looks to me like we're taking the allowance and than grabbing that same allowance out of the list by id

Reply:
Oh good question! This is because I need the user associated with the email but the allowance is reduced to not include this in the AllowanceList. So the allowance that is passed in here doesn't actually contain the user/username/email/userId. See here: https://github.com/edx/frontend-app-exams-dashboard/blob/a2da333a195943e51cc841b07dda6a2c18651c28/src/pages/ExamsPage/components/AllowanceList.jsx#L25

Reply:
ah interesting, seems to me the easiest path here is to make sure that allowance object does have all the properties